pkg/virtctl/vm/vm.go

```diff
@@ -433,229 +108,35 @@ func getDryRunOption(dryRun bool) []string {
 	return nil
 }
 
-func expandArgs() cobra.PositionalArgs {
-	return func(cmd *cobra.Command, args []string) error {
-		if filePath == "" && vmName == "" {
-			return fmt.Errorf("error invalid arguments - VirtualMachine name or file must be provided")
-		}
-
-		if outputFormat != YAML && outputFormat != JSON {
-			return fmt.Errorf("error not supported output format defined: %s", outputFormat)
-		}
-
-		return nil
-	}
-}
-
-func readVMFromFile(filePath string) (*v1.VirtualMachine, error) {
-	vm := &v1.VirtualMachine{}
-
-	readFile, err := os.ReadFile(filePath)
-	if err != nil {
-		return nil, fmt.Errorf("error reading file %+w", err)
-	}
-
-	err = yaml.Unmarshal(readFile, vm)
-	if err != nil {
-		return nil, fmt.Errorf("error decoding VirtualMachine %+w", err)
-	}
-
-	return vm, nil
-}
-
-func applyOutputFormat(outputFormat string, expandedVm *v1.VirtualMachine) (string, error) {
-	var formatedOutput []byte
-	var err error
-
-	switch outputFormat {
-	case JSON:
-		formatedOutput, err = json.MarshalIndent(expandedVm, "", " ")
-	case YAML:
-		formatedOutput, err = yaml.Marshal(expandedVm)
-	}
-
-	if err != nil {
-		return "", err
-	}
-
-	return string(formatedOutput), nil
+type vmOptions struct {
+	vmiName      string
+	namespace    string
+	dryRunOption []string
+	virtClient   kubecli.KubevirtClient
 }
 
-func expandVirtualMachine(namespace string, virtClient kubecli.KubevirtClient, o *Command) error {
-	var expandedVm *v1.VirtualMachine
+func BeforeRun(args []string, clientConfig clientcmd.ClientConfig) (*vmOptions, error) {
+	vars := vmOptions{}
 	var err error
 
-	if vmName != "" {
-		expandedVm, err = virtClient.VirtualMachine(namespace).GetWithExpandedSpec(vmName)
-		if err != nil {
-			return fmt.Errorf("error expanding VirtualMachine - %s in namespace - %s: %w", vmName, namespace, err)
-		}
-	} else {
-		vm, err := readVMFromFile(filePath)
-		if err != nil {
-			return err
-		}
-
-		expandedVm, err = virtClient.ExpandSpec(namespace).ForVirtualMachine(vm)
-		if err != nil {
-			return fmt.Errorf("error expanding VirtualMachine - %s in namespace - %s: %w", vm.Name, namespace, err)
-		}
-	}
-
-	output, err := applyOutputFormat(outputFormat, expandedVm)
-	if err != nil {
-		return err
-	}
-
-	o.cmd.Print(output)
-	return nil
-}
-
-func (o *Command) Run(args []string) error {
 	if len(args) != 0 {
```

Comment:
Leave it up to each cmd to get the vmiName from the args? It's not applicable to all cmds, right?